File: leetcode/golang/485.max-consecutive-ones.go
func findMaxConsecutiveOnes(nums []int) int {
	res := 0
	length := 0
	for _, num := range nums {
		if num == 1 {
			length++
		} else {
			res = max(res, length)
			length = 0
		}
	}
	res = max(res, length)
	return res

}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: leetcode/python3/485.max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0
        count_max = 0
        n = len(nums)
        for i in range(n):
            if nums[i] == 1:
                count += 1
                if count > count_max:
                    count_max = count
            else:
                count = 0
        return count_max
